src/azure-cli/azure/cli/command_modules/sqlvm/custom.py

```diff
@@ -392,6 +393,64 @@ def sqlvm_update(cmd, instance, sql_virtual_machine_name, resource_group_name, s
     return instance
 
 
+# pylint: disable=unused-argument
+def sqlvm_enable_azure_ad_auth(client, cmd, sql_virtual_machine_name, resource_group_name, msi_client_id=None, skip_client_validation=None):
+    ''' Enable Azure AD authentication on a SQL virtual machine.
+
+        :param cmd: The CLI command.
+        :type cmd: AzCliCommand.
+        :param instance: The Sql Virtual Machine instance.
+        :type instance: SqlVirtualMachine.
+        :param resource_group_name: The resource group name
+        :type resource_group_name: str.
+        :param msi_client_id: The clientId of the managed identity used in Azure AD authentication.
+                              None means system-assigned managed identity
+        :type: str.
+        :param skip_client_validation: Whether to skip the client side validation. The server side validation always happens.
+                                       This parameter is used in the validation and ignored here.
+        :type: bool.
+
+        :return: The updated Sql Virtual Machine instance.
+        :rtype: SqlVirtualMachine.
+    '''
+
+    sqlvm_object = client.get(resource_group_name, sql_virtual_machine_name)
+
+    if sqlvm_object.server_configurations_management_settings is None:
+        sqlvm_object.server_configurations_management_settings = ServerConfigurationsManagementSettings()
+
+    sqlvm_object.server_configurations_management_settings.azure_ad_authentication_settings = AADAuthenticationSettings(client_id=msi_client_id if msi_client_id else '')
+
+    # Since it's a running operation, we will do the put and then the get to display the instance.
+    LongRunningOperation(cmd.cli_ctx)(sdk_no_wait(False, client.begin_create_or_update,
+                                                  resource_group_name, sql_virtual_machine_name, sqlvm_object))
+
+    return client.get(resource_group_name, sql_virtual_machine_name)
```

Comment:
Will `client.begin_create_create_or_update` return a LROPoller? I think cli core can handle the poller automatically you don't have to do the get yourself.